File: Backend/finki-community-server/src/main/java/com/sorsix/finkicommunity/domain/responses/user_details/UserDetailsResponse.java
package com.sorsix.finkicommunity.domain.responses.user_details;

import com.sorsix.finkicommunity.domain.enums.Role;
import java.util.ArrayList;
import java.util.List;

public class UserDetailsResponse {
    public long userId;
    public String username;
    public String email;
    public String firstName;
    public String lastName;
    public Character sex;
    public long birthdate;
    public Role role;
    public boolean isFollowing;

    public int numberOfPosts;
    public int numberOfPostsLiked;
    public int numberOfFollowers;
    public int numberOfFollowings;

    public List<UserDetailsPost> userDetailsPosts = new ArrayList<>();
    public List<UserDetailsFollow> userDetailsFollowers = new ArrayList<>();
    public List<UserDetailsFollow> userDetailsFollowings = new ArrayList<>();
    public List<UserDetailsPost> userDetailsPostsLiked = new ArrayList<>();
}

File: Frontend/finki-community-client/src/app/Components/Views/user-details/models/classes/user-details-response.ts
import {UserDetailsFollow} from './user-details-follow';
import {UserDetailsPost} from './user-details-post';

export class UserDetailsResponse {
  constructor(
        public userId: number,
        public username: string,
        public email: string,
        public firstName: string,
        public lastName: string,
        public sex: string,
        public role: string,
        public isFollowing: boolean,

        public numberOfPosts: number,
        public numberOfFollowers: number,
        public numberOfFollowings: number,
        public numberOfPostsLiked: number,
        public userDetailsPosts: Array<UserDetailsPost>,
        public userDetailsFollowers: Array<UserDetailsFollow>,
        public userDetailsFollowings: Array<UserDetailsFollow>,
        public userDetailsPostsLiked: Array<UserDetailsPost>
  ) {
  }
}
